datafusion/sqllogictest/test_files/dynamic_file.slt

```diff
@@ -25,9 +25,170 @@ SELECT * FROM '../core/tests/data/partitioned_table_arrow/part=123' ORDER BY f0;
 1 foo true
 2 bar false
 
-# dynamic file query doesn't support partitioned table
-statement error DataFusion error: Error during planning: table 'datafusion.public.../core/tests/data/partitioned_table_arrow' not found
-SELECT * FROM '../core/tests/data/partitioned_table_arrow' ORDER BY f0;
```

Comment:
It's what I mentioned in https://github.com/apache/datafusion/issues/12671#issuecomment-2381291431
I think it isn't a dynamic file catalog issue, so I removed it from the test.